vcenterreceiver: Adds vcenter.datacenter.name attributes to all resource types

StefanKurek opened the pull request:
**Description:** <Describe what has changed.>
Adds `vcenter.datacenter.name` attribute for every resource type. This ensures uniqueness in cases where more than one datacenter exists.

**Link to tracking Issue:** <Issue number if applicable>
#32531 

**Testing:** <Describe what testing was performed and which tests were added.>
Unit/integration tests updated and tested. Local environment tested.

**Documentation:** <Describe the documentation added.>
Documentation regenerated.

StefanKurek committed:
vcenterreceiver adds vcenter.datacenter.name attributes to all resource types

djaglowski commented:
To clarify, this attribute would be applied to all resources emitted by the receiver, because all the components described are always associated with one data center. Is that right?

If so, this makes sense to me. We need to add it first as an optional resource attribute and can enable it by default in a future version. Same mechanism as https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/24071#issuecomment-2066775491 should be used.

djaglowski commented on receiver/vcenterreceiver/metadata.yaml:
```diff
@@ -7,10 +7,14 @@ status:
+resource_attributes:  
+  vcenter.datacenter.name:
+    description: The name of the vCenter datacenter.
+    enabled: true
```
It'll need to be optional initially.
```suggestion
    enabled: false
```

StefanKurek replied:
@djaglowski Change should be implemented

djaglowski commented on receiver/vcenterreceiver/testdata/metrics/expected.yaml:
```diff
@@ -1,6 +1,9 @@
 resourceMetrics:
   - resource:
       attributes:
+        - key: vcenter.datacenter.name
```
Since it'll be optional by default, I'd expect this file to be unchanged. However, it would be great to have another test which enables it, so we can see that it's producing expected values.

StefanKurek committed:
vcenterreceiver makes datacenter Resource Attr disabled by default

vcenterreceiver `vcenter.datacenter.name` resource attribute is disabled by default.
adds warning message that it will be enabled starting in release v.0.100.0.

StefanKurek commented:
@djaglowski should be ready for your review now. Also added the warning in the metadata for the new resource attribute.

djaglowski commented on receiver/vcenterreceiver/metadata.yaml:
```diff
@@ -7,10 +7,16 @@ status:
+    enabled: false
+    type: string
+    warnings:
+      if_enabled_not_set: "this attribute will be enabled by default starting in release v0.100.0"
```
v0.100.0 is the next release at this point so v0.101.0 would be the soonest possible to enable it by default.
```suggestion
      if_enabled_not_set: "this attribute will be enabled by default starting in release v0.101.0"
```

djaglowski committed:
Update receiver/vcenterreceiver/metadata.yaml

StefanKurek committed:
vcenterreceiver runs make generate after updating metadta

StefanKurek commented:
@djaglowski should have all conflicts fixed now.
